### DISCUSSION TITLE
Can we perform pseudo-relevance feedback with BM25S?

### DISCUSSION THREAD

**Discussion by roynirmal** [xhluca/bm25s#71] [Q&A] [answered]
Thanks for the amazing tool!

Is there any way we a perform pseudo-relevance feedback using your implementation? Or any suggestions on how I can do the same?
https://nlp.stanford.edu/IR-book/html/htmledition/pseudo-relevance-feedback-1.html

**Comment by xhluca** [comment, marked as answer]
Sorry, i'm not familiar with the concept, and it's challenging for me to work out the math wrt non-occurrence matrix and score shifting. If you work it out, feel free to share it here!

Regarding building bm25 scorers in general, you can implement your own idf scorer in a fork and add it here: https://github.com/xhluca/bm25s/blob/f77e90d79e9a3b2303d12ec4fbfd35ffa9f5e0d5/bm25s/scoring.py#L206

and create your own tf component scorer and add it here: https://github.com/xhluca/bm25s/blob/f77e90d79e9a3b2303d12ec4fbfd35ffa9f5e0d5/bm25s/scoring.py#L146

then, in the bm25s, you can select it by using the name of your custom bm25 implementation.
